### PR TITLE
Add instructions on how to verify sources for PMC members

### PR DESCRIPTION
**Pull request by potiuk**

### CODE REVIEW

File: dev/README_RELEASE_CLIENT.md [hunk 1]
```diff
@@ -283,6 +283,20 @@ See Airflow process documented [here](https://github.com/apache/airflow/blob/mas
 ## Signature check
 See Airflow process documented [here](https://github.com/apache/airflow/blob/master/dev/README_RELEASE_AIRFLOW.md#signature-check).
 
+## Sources check
+
+The code of the Python Client is generated using Open API generator and the generated code is committed to
```
Comment (hussein-awala) [inline dev/README_RELEASE_CLIENT.md:288]:
```suggestion
The code of the Python Client is generated using OpenAPII generator and the generated code is committed to
```

File: dev/README_RELEASE_CLIENT.md [hunk 2]
```diff
@@ -283,6 +283,20 @@ See Airflow process documented [here](https://github.com/apache/airflow/blob/mas
...
+the repository, therefore the Source code check should consist of two steps:
+
+1. Checkout the release tag of the "airflow-client-python"
+2. Follow the [Release process](#release-process) above and generate client's code above
+3. Run ``git diff`` to see the differences - review them. Generally the code generated should only contain
+   small differences regarding authentication.
+4. git reset --hard <RELEASE_TAG>
+
+Compare the sources with the packaged sources following the process describd in Airflow:
```
Comment (hussein-awala) [inline dev/README_RELEASE_CLIENT.md:297]:
```suggestion
Compare the sources with the packaged sources following the process described in Airflow:
```

File: dev/README_RELEASE_CLIENT.md [hunk 3]
```diff
@@ -283,6 +283,20 @@ See Airflow process documented [here](https://github.com/apache/airflow/blob/mas
...
 See Airflow process documented [here](https://github.com/apache/airflow/blob/master/dev/README_RELEASE_AIRFLOW.md#signature-check).
 
+## Sources check
+
+The code of the Python Client is generated using Open API generator and the generated code is committed to
+the repository, therefore the Source code check should consist of two steps:
+
+1. Checkout the release tag of the "airflow-client-python"
+2. Follow the [Release process](#release-process) above and generate client's code above
+3. Run ``git diff`` to see the differences - review them. Generally the code generated should only contain
```
Comment (hussein-awala) [inline dev/README_RELEASE_CLIENT.md:293]:
```suggestion
3. Run ``git diff`` to see the differences - review them. Generally, the code generated should only contain
```

**Review by hussein-awala — APPROVED** [review]

**Review by pierrejeambrun — APPROVED** [review]

**Comment by pierrejeambrun** [comment]
Ultimately, this could be handled by breeze, like release-management, client release, create-rc. And we would not have to do anything else than running a simple command.

**Comment by potiuk** [comment]
> Ultimately, this could be handled by breeze, like release-management, client release, create-rc. And we would not have to do anything else than running a simple comman, minimizing the risk of error.

Agree. And run it on CI to verify that it works - we kind-of-doing-it  already, we have open-api-client generation - but it does not use breeze. but having both "build" and "verify-provenence" steps as a ready to run command in Breeze would be a nice thing to automate.

**Comment by potiuk** [comment]
Of course there is always a risk of over-automating AKA "who checks the checks" and something to keep it running, but once we run it in CI and make some proper diagnostics ouput to show the steps being executed and output (diff) + something that I also just did - run it on WRONG version and see differences to make sure that it **actually** detects difference, should be good enough.

**Pull request merged**